---
layout: post
type: project
title: "Sleuth"
date:   2016-05-24 2:35PM
location: "UC Berkeley"
categories: [RNA-Seq, Computational Biology]
link: http://pachterlab.github.io/sleuth/
git-link: https://github.com/pachterlab/sleuth
description: >
    Sleuth is an RNA-seq tool to perform differential expression analysis of quantified reads from
    kallisto
    and can display the analysis in a web browser using
    Shiny by RStudio.
    It uses kallisto's bootstrap estimates to implement novel statistical
    methods for differential analysis.
background: >
    RNA-Sequencing, or RNA-seq for short, was developed in 2008 and is a
    type of study that aims, at a high level, to understand gene expression
    within an organism. RNA-seq is used to analyze the transcriptome,
    which, unlike the genome, is constantly changing. Among other things, RNA-seq
    helps us understand what types of conditions can cause cells to express
    certain genes (differential analysis), and helps us discover new
    transcripts and explore their purpose.<br /> <br />

    In an RNA-seq experiment, RNA from an organism is chemically prepared and
    fed through a sequencer. The sequencer generates "reads" in the millions:
    each read is a chunk of RNA-molecule that came from the original organism.
    A computer program, such as <a target="_blank" href="https://pachterlab.github.io/kallisto">kallisto</a>,
    is then used to find which genes these reads came from. This process is called
    loosely called "alignment." The more reads that mapped to a particular gene,
    the more that gene was being expressed at the time of the sequencing.
    Due to the sheer volume of data, there are many difficult computational
    problems involved with RNA-seq alignment, many of which are discussed in the
    <a target="_blank" href="http://www.nature.com/nbt/journal/v34/n5/full/nbt.3519.html">kallisto paper</a>.
    <br /><br />

    After the reads have been aligned, another computer program, such as
    <a target="_blank" href="http://pachterlab.github.io/sleuth/">sleuth</a>, is used to
    determine, statistically speaking, which genes were most
    <i>differentially expressed</i> across a condition. Whether or not a gene
    was significantly different across a condition is typically determined by
    p-value or q-value. From these analyses, computational biologists
    draw conclusions about which genes are important to which conditions; an
    example is <a target="_blank" href="http://www.ncbi.nlm.nih.gov/pubmed/25258427">Sharma et al.</a>,
    a paper discussing genes that are important when cells die.

    <br /><br />
    The above Sharma et al. paper is particularly important to me because
    it was the first RNA-seq paper I ever read!

my_role: >
    My role in this project was twofold. First, I designed several of the data
    visualizations present in the Sleuth live app (see the
    <a target="_blank" href="http://lair.berkeley.edu/ellahi">example anlysis</a>).
    I designed the gene table and the gene viewer, which display
    the variation across conditions of given genes. I created the
    transcript heat map plot, which uses a hierarchical clustering algorithm
    to group transcripts with similar expression patterns together.
    Clustered heat maps are useful to chunk large data sets into related groups.
    I implemented the volcano plot, a classic plot in statistics with
    significance of expression on the y-axis and fold-change on the x-axis.
    This plot is especially useful to find outlying transcripts that have
    high fold-change but low significance. My implementation of the plot is reactive,
    and lets a user highlight particular data points and analyze those points
    in a stand-alone table. <br /><br />

    Second, I made it possible for high-volume Sleuth objects to be shared online. Initially,
    the Sleuth objects and anlyses were too large to be shared online because they
    stored statistical bootstraps of the original data to generate the
    data visualizations live in the browser. I refactored the Sleuth code to
    pre-compute key statistcs needed to display the plots. This allowed
    Sleuth to abandon the cumbersome bootstrap data after pre-computation. This effectively
    reduced the size of the analyses objects by orders of magnitude, for the
    first time allowing RNA-seq analyses to shared live in a browser. Before Sleuth, RNA-seq experiments were only available as raw data in the
    <a target="_blank" href="http://www.ncbi.nlm.nih.gov/sra">Short Read Archive</a> and needed several days and multiple
    cores to re-analyze. <br /> <br />

    I am now working on compiling a database of re-analyzed RNA-seq experiments that will
    shortly be a live webpage. For now, check out the <a target="_blank" href="https://github.com/pachterlab/lair">
    Github repo for the database here</a>.


---
Sleuth is an RNA-seq tool that performs differential expression analysis on quantified reads from
<a target="_blank" href="https://pachterlab.github.io/kallisto">kallisto</a>.
Sleuth is novel both in its use of statistical bootstraps to understand
differential gene expression, and in its use of
<a target="_blank" href="http://shiny.rstudio.com">Shiny</a> to provide in-browser, interactive
analyses of RNA-seq experiments.
If you are interested in using sleuth, see the
<a target="_blank" href="http://pachterlab.github.io/sleuth/">sleuth homepage</a>.
If you would like to see an example of a live sleuth analysis, see
<a target="_blank" href="http://lair.berkeley.edu/">the bear's lair</a>.
<br /><br />
Sleuth was developed by the <a target="_blank" href="http://pachterlab.github.io">Pachter Lab</a>
at UC Berkeley.
The source code for sleuth is available on Github at the
<a target="_blank" href="https://github.com/pachterlab/sleuth"><span class="icon icon--github">
{% include icon-github.svg %}</span><span class="username">Sleuth Repo</span></a>.
