---
layout: post
type: project
title: "The bear's lair"
date:   2016-05-28 7:25AM
location: "UC Berkeley"
categories: [RNA-Seq, Computational Biology, Web]
link: http://pachterlab.github.io/lair/
git-link: https://github.com/pachterlab/lair/
description: >
    <a target="_blank" href="http://pachterlab.github.io/lair/">The bear's lair</a> is a website
    that hosts interactive reanalyses of publicly available RNA-seq data. The
    analyses are automatically run by a back-end Snakemake buildsystem and
    performed by kallisto and sleuth.
    The website promotes verifiability of RNA-seq experiements.

background: >
    This website is directly related to sleuth, another project I've contributed to.
    Sleuth is a differential analysis tool and is the first of such tools to
    be able to host its analysis live in a browser. If sleuth could turn RNA-seq
    analyses into webpages, the natural next step was to create a database that
    housed all of these analyses in a single place â€“ thus, the bear's lair was created.
    <br /> <br />

    The goal of the website is to have all authors of RNA-seq papers submit their raw data
    to the website, and then have our automated server process and upload their results.
    In doing so, individual authors increase their paper's credibility, and the
    bioinformatics community benefits from more transparency. Furthermore,
    by having the analysis live, users can continually discover new results
    that may have been missed in the original publication.


my_role: >
    I designed the entirety of the website. The website is build using
    Jekyll and Bootstrap and hosted by Github Pages. The table in the analysis tab,
    which features the papers we analyzed, is powered by the jQuery DataTables plug-in. <br /><br />

    I also designed the back-end automated workflow that analyzes RNA-seq data
    and uploads the results to our server. The workflow is made using Snakemake,
    a build-system created from python and frequenty used in computational biology.
    The workflow is depicted below:
    <img src="/_images/workflow.pdf" style="width: 100%; margin: auto; border-radius: 5%;" />



---

Reproducibility is critical to experiemental results. To this end, many
RNA-seq papers archive their data in the Sequence Read Archive. This data is theoretically
enough to reproduce the results of the original paper; doing so is typically infeasable
due to the computational power required to process the results and the computational
biology background needed to select the right computational tools. <br /><br />

In order to make RNA-seq results transparent,
we (the <a href="http://pachterlab.github.io/">Pachter Lab</a> and I)
created <a target="_blank" href="http://pachterlab.github.io/lair/">the bear's lair</a>.
It is a website that hosts interactive reanalyses of publicly available RNA-seq data. The
analyses are automatically run by a back-end Snakemake buildsystem and
performed by
<a target="_blank" href="http://pachterlab.github.io/kallisto/">kallisto</a> and
<a target="_blank" href="http://pachterlab.github.io/sleuth/">sleuth</a>.
The website allows anyone to inspect the results of RNA-seq experiments without
needing computing power or extensive knowledge of alignment/analysis tools. <br /><br />

You can take a look at the <a target="_blank" href="https://github.com/pachterlab/lair"><span class="icon icon--github">
{% include icon-github.svg %}</span><span class="username">bear's lair repo</span></a>
to see the source. The papers in the analysis tab are hosted in the <a target="_blank" href="https://github.com/pachterlab/bears_analyses"><span class="icon icon--github">
{% include icon-github.svg %}</span><span class="username">bears_analyses repo</span></a>.
